Specs/Scene/parseFeatureMetadataSpec.js

```diff
@@ -0,0 +1,286 @@
+      source: {
+        arrayBufferView: new Uint8Array([5]),
+        width: 1,
+        height: 1,
+      },
+    });
+
+    var textures = {
+      0: texture0,
+      1: texture1,
+    };
+
+    var extension = {
+      schema: featureTexturesSchema,
+      featureTextures: {
+        mapTexture: {
+          class: "map",
+          properties: {
+            color: {
+              channels: "rgb",
+              texture: {
+                index: 0,
+                texCoord: 0,
+              },
+            },
+            intensity: {
+              channels: "a",
+              texture: {
+                index: 0,
+                texCoord: 0,
+              },
+            },
+          },
+        },
+        orthoTexture: {
+          class: "ortho",
+          properties: {
+            vegetation: {
+              channels: "r",
+              texture: {
+                index: 1,
+                texCoord: 1,
+              },
+            },
+          },
+        },
+      },
+    };
+
+    var metadata = parseFeatureMetadata({
+      extension: extension,
+      schema: new MetadataSchema(featureTexturesSchema),
+      textures: textures,
+    });
+
+    var mapClass = metadata.schema.classes.map;
+    var orthoClass = metadata.schema.classes.ortho;
+
+    expect(mapClass.id).toBe("map");
+    expect(orthoClass.id).toBe("ortho");
+
+    var mapTexture = metadata.getFeatureTexture("mapTexture");
+    var orthoTexture = metadata.getFeatureTexture("orthoTexture");
+
+    expect(mapTexture.class).toBe(mapClass);
+    expect(orthoTexture.class).toBe(orthoClass);
+
+    texture0.destroy();
+    texture1.destroy();
+    context.destroyForSpecs();
+  });
+
+  it("parses extension with statistics", function () {
+    var statistics = {
+      classes: {
+        tree: {
+          count: 100,
+          properties: {
+            height: {
+              min: 10.0,
+              max: 20.0,
+            },
+          },
+        },
+      },
+    };
+    var extension = {
+      statistics: statistics,
+    };
+    var metadata = parseFeatureMetadata({
+      extension: extension,
+      schema: new MetadataSchema(featureTexturesSchema),
+    });
+    expect(metadata.statistics).toBe(statistics);
+  });
+
+  it("parses extension with extras", function () {
+    var extras = {
+      description: "Extra",
+    };
+    var extension = {
+      extras: extras,
+    };
+    var metadata = parseFeatureMetadata({
+      extension: extension,
+      schema: new MetadataSchema(featureTexturesSchema),
+    });
+    expect(metadata.extras).toBe(extras);
+  });
+
+  it("parses extension with extensions", function () {
+    var extensions = {
+      EXT_other_extension: {},
+    };
+    var extension = {
+      extensions: extensions,
+    };
+    var metadata = parseFeatureMetadata({
+      extension: extension,
+      schema: new MetadataSchema(featureTexturesSchema),
+    });
+
+    expect(metadata.extensions).toBe(extensions);
+  });
+});
```

Comment:
Since `Textures` are created in here this needs to be marked as a `"WebGL"` test (I forgot to do this originally)